src/sentry/integrations/slack/webhooks/action.py

```diff
@@ -280,10 +301,64 @@ def open_resolve_dialog(self, slack_request: SlackActionRequest, group: Group) -
             "trigger_id": slack_request.data["trigger_id"],
         }
         slack_client = SlackClient(integration_id=slack_request.integration.id)
-        try:
-            slack_client.post("/dialog.open", data=payload)
-        except ApiError as e:
-            logger.error("slack.action.response-error", extra={"error": str(e)})
+
+        if use_block_kit:
+            # XXX(CEO): the second you make a selection (without hitting Submit) it sends a slightly different request
+            formatted_resolve_options = []
+            for text, value in RESOLVE_OPTIONS.items():
+                formatted_resolve_options.append(
+                    {
+                        "text": {
+                            "type": "plain_text",
+                            "text": text,
+                            "emoji": True,
+                        },
+                        "value": value,
+                    }
+                )
+
+            modal_payload = {
+                "type": "modal",
+                "title": {"type": "plain_text", "text": "Resolve Issue"},
+                "blocks": [
+                    {
+                        "type": "section",
+                        "text": {"type": "mrkdwn", "text": "Resolve in"},
+                        "accessory": {
+                            "type": "static_select",
+                            "initial_option": {
+                                "text": {
+                                    "type": "plain_text",
+                                    "text": "Immediately",
+                                    "emoji": True,
+                                },
+                                "value": "resolved",
+                            },
+                            "options": formatted_resolve_options,
+                            "action_id": "static_select-action",
+                        },
+                    }
+                ],
+                "close": {"type": "plain_text", "text": "Cancel"},
+                "submit": {"type": "plain_text", "text": "Resolve"},
+                "private_metadata": callback_id,
+                "callback_id": callback_id,
+            }
+            try:
+                payload = {
+                    "view": json.dumps(modal_payload),
+                    "trigger_id": slack_request.data["trigger_id"],
+                }
+                headers = {"content-type": "application/json; charset=utf-8"}
+                slack_client.post("/views.open", data=json.dumps(payload), headers=headers)
+            except ApiError as e:
+                logger.error("slack.action.response-error", extra={"error": str(e)})
+
+        else:
+            try:
+                slack_client.post("/dialog.open", data=payload)
+            except ApiError as e:
+                logger.error("slack.action.response-error", extra={"error": str(e)})
```

Comment:
Hmm... Didn't you get flake8 whining about this? 

Inside except it should be `.exception()` 
https://github.com/adamchainz/flake8-logging?tab=readme-ov-file#log005-use-exception-within-an-exception-handler

Reply:
no whining, but I'll update it